contrib/NeonVM/cell.c

```diff
@@ -125,12 +125,48 @@ Cell *cell_createArrayCell(uint32_t iElements)
     if (c->array == NULL) {
         fatal_error("Unable to allocate memory for array.");
     }
-    for (uint32_t i = 0; i < c->array_size; i++) {
+    for (uint64_t i = 0; i < c->array_size; i++) {
         cell_resetCell(&c->array[i]);
     }
     return c;
 }
 
+Cell *cell_arrayIndexForRead(Cell *c, size_t i)
+{
+    if (c->type == None) {
+        c->type = Array;
+    }
+    assert(c->type == Array);
+    if (!c->array) {
+        c->array = malloc(sizeof(Cell) * i);
+        c->array_size = i;
+        if (c->array == NULL) {
+            fatal_error("Unable to allcoate memory for Read Array.");
+        }
+    }
+    return &c->array[i];
+}
+
+Cell *cell_arrayIndexForWrite(Cell *c, size_t i)
+{
+    if (c->type == None) {
+        c->type = Array;
+    }
+    assert(c->type == Array);
+    if (!c->array) {
+        c->array = malloc(sizeof(Cell) * i);
```

Comment:
Why are you allocating space for i elements here? Even if this path is taken, you're going to realloc next anyway.